contractrequester/contractrequester.go

```diff
@@ -174,6 +174,13 @@ func (cr *ContractRequester) Call(ctx context.Context, inMsg insolar.Payload) (i
 	if msg.Request.Nonce == 0 {
 		msg.Request.Nonce = randomUint64()
 	}
+	if msg.PulseNumber == 0 {
+		pulseObject, err := cr.PulseAccessor.Latest(ctx)
```

Comment:
Is it necessary? Looks like shady logic.

Reply:
We cannot be sure that someone will send us message with correct PulseNumber, so this check is necessary.